### ISSUE TITLE
SemanticVersion.TryParse() vs. TagPrefix

### ISSUE THREAD

**Issue by Tungsten78** [GitTools/GitVersion#495]
First off, great work on this tool!  You guys have put together something great.

I've been trying to integrate GitVersion (v3.0.0-beta.4) into our build process and have been running into issues with respect to the tag strategies.  Our build server tags our repository with a build tag of the form "build/product/branch/1.2.3".  This allows our team to easily track commits back to the most recent build.

The issue stems from GitVersion using these tags within TrackMergeTargetBaseVersionStrategy and TaggedCommitVersionStrategy.   Given the default prefix of [vV] we expected that our tags would be ignored.

I was looking at the commit that added TagPrefix (https://github.com/JakeGinnivan/GitVersion/commit/31d42088106e6ca9100260c74e18b8bfb0022f65) and it seems from the unit tests that the intent was to be more flexible when parsing tags (instead of only allowing an optional v).  What's odd is that regardless of what we use for TagPrefix the result is always successful.

Consider (tagprefix changed): 
```c#
    [Test]
    public void CanSpecifyTagPrefixes()
    {
        using (var fixture = new EmptyRepositoryFixture(new Config { TagPrefix = "SomethingOfftheWall" }))
        {
            const string TaggedVersion = "version-1.0.3";
            fixture.Repository.MakeATaggedCommit(TaggedVersion);
            fixture.Repository.MakeCommits(5);
            fixture.AssertFullSemver("1.0.4+5");
        }
    }
```
Here's a failed test to illustrate what I was expecting:
```c#
[Test]
    public void AreTagsNotAdheringToTagPrefixIgnored()
    {
        using (var fixture = new EmptyRepositoryFixture(new Config { TagPrefix = "" }))
        {
            string TaggedVersion = "version-1.0.3";
            fixture.Repository.MakeATaggedCommit(TaggedVersion);
            fixture.Repository.MakeCommits(5);

            fixture.AssertFullSemver("0.1.0+5");    //Fallback version + 5 commits since tag

            TaggedVersion = "bad/1.0.3";
            fixture.Repository.MakeATaggedCommit(TaggedVersion);

            fixture.AssertFullSemver("0.1.0+6");   //Fallback version + 6 commits since tag
        }
    }
```

I've made a few edits (https://github.com/GitTools/GitVersion/compare/master...Tungsten78:AddressTagParsing) to the regular expression to modify this behavior.  The changes involve adding start (^) and end ($) constraints to the regular expressions to ensure that the match applies to the entire tag string.  

Also, whether the TagPrefix is optional is now specified in the prefix itself (ie: TagPrefix= "[vV]|").  This was made the default configuration.

**Comment by Tungsten78** [comment]
Edit:
I revisited by making the default TagPrefix optional again  (https://github.com/Tungsten78/GitVersion/commit/f459594a76774fa84cf01d0c9154455cf60748ba) - also more unit tests.

This should not have impact on those already depending on its behavior.

**Comment by JakeGinnivan** [comment]
I have opened a PR, the changes look good. If the build passes I will merge

Thanks for the fixes!

**Comment by JakeGinnivan** [comment]
Merged into master via 50833b4900ad653ddabfade567d58c52ef669110

Thanks!
